tests/test_csv_splitter.py

```diff
@@ -0,0 +1,72 @@
+from main import validate_file_length, is_input_valid, get_file_chunks, create_files
+import pytest
+import os
+from mock_data import CHUNKS, CSV_DATA, TITLE
+from typing import Callable
+from utils_types import StringGeneratorType
+
+
+def create_temp_file_fixture(file_extension: str) -> Callable[[], StringGeneratorType]:
+    @pytest.fixture
+    def temp_file() -> StringGeneratorType:
+        file_name = f"test.{file_extension}"
+        with open(file_name, "w") as test_file:
+            test_file.write(CSV_DATA)
+        yield file_name
+        os.remove(file_name)
+
+    return temp_file
+
+
+temp_csv_file = create_temp_file_fixture(file_extension="csv")
+temp_txt_file = create_temp_file_fixture(file_extension="txt")
+
+
+@pytest.fixture
+def create_files_from_chunks():
+    create_files(chunks=CHUNKS, output_file_name="out")
+    yield
+    os.remove("out_1.csv")
+    os.remove("out_2.csv")
+
+
+def test_validate_file_length_success(temp_csv_file):
+    assert validate_file_length(input_file=temp_csv_file, row_limit=3) is True
+
+
+def test_validate_file_length_should_fail_row_limit_too_high(temp_csv_file):
+    assert validate_file_length(input_file=temp_csv_file, row_limit=8) is False
+
+
+def test_is_input_valid_success(temp_csv_file):
+    assert is_input_valid(input_file=temp_csv_file, row_limit=5) is True
+
+
+def test_is_input_valid_should_fail_file_not_exist():
+    assert is_input_valid(input_file="non_existent_file.csv", row_limit=3) is False
+
+
+def test_is_input_valid_should_fail_not_csv(temp_txt_file):
+    assert is_input_valid(input_file=temp_txt_file, row_limit=3) is False
+
+
+def test_get_file_chunks(temp_csv_file):
+    chunks = get_file_chunks(temp_csv_file, 4)
```

Comment:
Again, let's not use magic numbers. I want to understand where each one of them came from.